app/src/main/java/fr/neamar/kiss/SettingsActivity.java

```diff
@@ -397,6 +398,18 @@ public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, Strin
             if (provider != null) {
                 provider.reload();
             }
+        } else if (key.equalsIgnoreCase("gestures-swipe-up")) {
+            String pref_value = prefs.getString("gestures-swipe-up", "apps");
+            ExperienceTweaks.updateGestureActionCache("up", pref_value);
```

Comment:
Although this does work; you can also add the four `gestures-swipe-` in this line: https://github.com/Neamar/KISS/blob/master/app/src/main/java/fr/neamar/kiss/SettingsActivity.java#L50

This way the main activity will be refreshed and you don't need to communicate with `ExperienceTweaks` :)

Reply:
(you also avoid the need for static variables, since they work weirdly in some Android devices: when memory gets low, static values can be problematic and force the system to kill the app completely)